src/resources/auth/middleware/Roles.ts

```diff
@@ -0,0 +1,219 @@
+			{ action: "DELETE:/attractions/:identifier/externallinks" },
+			{ action: "POST:/attractions/:identifier/archive" },
+			{ action: "POST:/attractions/:identifier/unarchive" },
+			{ action: "GET:/organizations/" },
+			{ action: "POST:/organizations/search" },
+			{ action: "GET:/organizations/:identifier" },
+			{ action: "GET:/events/" },
+			{ action: "POST:/events/" },
+			{ action: "POST:/events/bulk-create" },
+			{ action: "POST:/events/search" },
+			{ action: "GET:/events/:identifier" },
+			{ action: "PATCH:/events/:identifier" },
+			{ action: "PUT:/events/:identifier/locations" },
+			{ action: "DELETE:/events/:identifier/locations" },
+			{ action: "PUT:/events/:identifier/attractions" },
+			{ action: "DELETE:/events/:identifier/attractions" },
+			{ action: "PUT:/events/:identifier/organizer" },
+			{ action: "DELETE:/events/:identifier/organizer" },
+			{ action: "POST:/events/:identifier/reschedule" },
+			{ action: "POST:/events/:identifier/postpone" },
+			{ action: "POST:/events/:identifier/cancel" },
+			{ action: "POST:/events/:identifier/archive" },
+			{ action: "POST:/events/:identifier/unarchive" },
+			{ action: "POST:/events/:identifier/duplicate" },
+			{ action: "GET:/locations/" },
+			{ action: "POST:/locations/" },
+			{ action: "POST:/locations/bulk-create" },
+			{ action: "POST:/locations/search" },
+			{ action: "GET:/locations/:identifier" },
+			{ action: "PATCH:/locations/:identifier" },
+			{ action: "POST:/locations/:identifier/manager" },
+			{ action: "DELETE:/locations/:identifier/manager" },
+			{ action: "POST:/locations/:identifier/open" },
+			{ action: "POST:/locations/:identifier/close" },
+			{ action: "POST:/locations/:identifier/permanentlyClose" },
+			{ action: "POST:/locations/:identifier/archive" },
+			{ action: "POST:/locations/:identifier/unarchive" },
+		],
+	},
+	{
+		role: member,
+		allowedRoutes: [
+			{ action: "GET:/attractions/" },
+			{ action: "GET:/attractions/:identifier" },
+			{ action: "GET:/organizations/" },
+			{ action: "POST:/organizations/search" },
+			{ action: "GET:/organizations/:identifier" },
+			{ action: "GET:/events/" },
+			{ action: "POST:/events/search" },
+			{ action: "GET:/events/:identifier" },
+			{ action: "GET:/locations/" },
+			{ action: "POST:/locations/search" },
+			{ action: "GET:/locations/:identifier" },
+		],
+	},
+	{
+		role: unassigned,
+		allowedRoutes: [],
+	},
+];
+
+export function getRoleByRoleName(roleName: string): Role {
+	const lowerCaseRoleName = roleName.toLowerCase();
+	const foundRole = Roles.find((role) => role.role === lowerCaseRoleName);
+	return foundRole || Roles.find((role) => role.role === unassigned)!;
+}
+
+export function checkPermissionForRole(roleName: string | undefined, action: string): boolean {
+	if (!roleName) {
+		return false;
+	}
+	const role = getRoleByRoleName(roleName);
+	return role.allowedRoutes.some((route) => route.action === action);
+}
```

Comment:
Hier würde ich auch strikter sein und `Role` statt `string` als Typ für den Parameter nehmen. Dann ist theoretisch auch kein lowercasing mehr nötig:

```suggestion
export function getRoleByRoleName(roleName: RoleType): Role {
	const foundRole = Roles.find((role) => role.role === roleName);
	return foundRole || Roles.find((role) => role.role === "unassigned")!;
}

export function checkPermissionForRole(roleName: RoleType | undefined, action: string): boolean {
	if (!roleName) {
		return false;
	}
	const role = getRoleByRoleName(roleName);
	return role.allowedRoutes.some((route) => route.action === action);
}
```

Reply:
ok